TypeCobol/Compiler/Parser/CodeElementBuilder/CobolCodeElementBuilder.cs

```diff
@@ -120,25 +117,46 @@ private void AddTokensConsumedAndDiagnosticsAttachedInContext(IList<Token> consu
         ////////////////////////////
 
         public override void EnterProgramIdentification(CodeElementsParser.ProgramIdentificationContext context) {
-			var program = new ProgramIdentification();
-			program.ProgramName = CobolWordsBuilder.CreateProgramNameDefinition(context.programNameDefinition());
-			if (context.COMMON() != null) {
-				program.Common = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.COMMON()));
-			}
-			if (context.INITIAL() != null) {
-				program.Initial = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.INITIAL()));
-			}
-			if (context.RECURSIVE() != null) {
-				program.Recursive = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.RECURSIVE()));
-			}
-			program.AuthoringProperties = CreateAuthoringProperties(context.authoringProperties());
-            
+            var program = new ProgramIdentification();
+            program.ProgramName = CobolWordsBuilder.CreateProgramNameDefinition(context.programNameDefinition());
+
+            if (context.COMMON() != null) {
+                program.Common = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.COMMON()));
+            }
+            if (context.INITIAL() != null) {
+                program.Initial = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.INITIAL()));
+            }
+            if (context.RECURSIVE() != null) {
+                program.Recursive = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.RECURSIVE()));
+            }
+            program.AuthoringProperties = CreateAuthoringProperties(context.authoringProperties());
+
             Context = context;
-			CodeElement = program;
+            CodeElement = program;
             IsDebuggingModeEnabled = false;
+
+            if (context.pgmIdPeriodSeparator == null)
+            {
+                //a dot doesn't follow PROGRAM-ID
+                if (CodeElement.Diagnostics == null) CodeElement.Diagnostics = new List<Diagnostic>();
+                CodeElement.Diagnostics.Add(new ParserDiagnostic("Dot expected after PROGRAM-ID", context.PROGRAM_ID().Symbol, null, MessageCode.Warning));
+            }
+            if (context.pgmIdDeclarPeriodSeparator == null)
+            {
+                //a dot doesn't follow PROGRAM-ID declaration
+                if (CodeElement.Diagnostics == null) CodeElement.Diagnostics = new List<Diagnostic>();
+                IToken previousToken = context.Stop; ;
```

Comment:
Maybe delete the extra ;

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1651/commits/40619858f34daad4a11c9a27f1b7f506e9f9a7f7